docs/plugins.md

```diff
@@ -1,3 +1,161 @@
 # Plugins
 
-TBD
+Plugin is a JavaScript package that extends built-in React Native CLI features. It can provide an array of additional commands to run or platforms to target.
+
+For example, `react-native-windows` package is a plugin that provides `react-native run-windows` command and `windows` platform.
+
+Details of this particular integration as well as how to provide an additional platform for React Native were described in a [`dedicated section`](./platforms.md) about platforms.
+
+## How does it work?
+
+Except for React Native dependencies, where configuration is implicit, each package needs to have a `react-native.config.js` at the root folder in order to be discovered by the CLI as a plugin.
+
+```js
+module.exports = {
```

Comment:
How about adding a short description to this file? 

> Below is an example of a project that exports a command named `foo-command` that logs "It worked" and exits.

It feels out of place without it.